examples/nlp/gpt2_text_generation_with_kerasnlp.py

```diff
@@ -0,0 +1,363 @@
+graph is compiled in the 1st run and re-used in the 2nd behind the scene.
+
+The quality of the generated text looks OK, but we can improved it via
+finetuning.
+"""
+
+"""
+## More on the GPT-2 model from KerasNLP
+
+While it may be sufficient to move on to the next step of finetuning the
+loaded model now, for the more curious you can better understand how generative
+LLMs are built by KerasNLP in this section.
+
+The code of GPT2 can be found
+[here](https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/models/gpt2/).
+Conceptually the `GPT2CausalLM` can be hierarchically broken down into several
+modules in KerasNLP, all of which have a *from_preset()* function that loads a
+pretrained model:
+
+- `keras_nlp.models.GPT2Tokenizer`: The tokenizer used by GPT2 model, which is a
+    [byte-pair encoder](https://huggingface.co/course/chapter6/5?fw=pt).
+- `keras_nlp.models.GPT2CausalLMPreprocessor`: the preprocessor used by GPT2
+    causal LM training. It does the tokenization along with other preprocessing
+    works such as creating the label and appending the end token.
+- `keras_nlp.models.GPT2Backbone`: the GPT2 model, which is a stack of
+    `keras_nlp.layers.TransformerDecoder`. This is usually just referred as
+    `GPT2`.
+- `keras_nlp.models.GPT2CausalLM`: wraps `GPT2Backbone`, it multiplies the
+    output of `GPT2Backbone` by embedding matrix to generate logits over
+    vocab tokens.
+"""
+
+"""
+## Finetune on Reddit dataset.
+
+Now you have the knowledge of the GPT-2 model from KerasNLP, you can take one
+step further to finetune the model so that it generates text in a specific
+style, short or long, strict or casual. In this tutorial, we will use reddit
+dataset for example.
+"""
+
+import tensorflow_datasets as tfds
+
+reddit_ds = tfds.load("reddit_tifu", split="train", as_supervised=True)
+
+"""
+Let's take a look inside sample data from the reddit TensorFlow Dataset. There
+are two features:
+
+- **__document__**: text of the post.
+- **__title__**: the title.
+
+"""
+
+for document, title in reddit_ds:
+    print(document.numpy())
+    print(title.numpy())
+    break
+
+"""
+In our case, we are performing next word prediction in a language model, so we
+only need the 'document' feature.
+"""
+
+train_ds = (
+    reddit_ds.map(lambda document, _: document)
+    .batch(32)
+    .cache()
+    .prefetch(tf.data.AUTOTUNE)
+)
+
+"""
+Now you can finetune the model using the familiar *fit()* function. Note that
+`preprocessor` will be automatically called inside `fit` method since
+`GPT2CausalLM` is a `keras_nlp.models.Task` instance.
+
+This step takes quite a bit of GPU memory and a long time if we were to train
+it all the way to a fully trained state. Here we just use part of the dataset
+for demo purposes.
+"""
+
+train_ds = train_ds.take(500)
+num_epochs = 1
+
+learning_rate = keras.optimizers.schedules.PolynomialDecay(
+    5e-5,
+    decay_steps=train_ds.cardinality() * num_epochs,
```

Comment:
should we show `power=1.0` explicitly, and mention this is a linear decay? i would find this confusing as a user

Reply:
Added a comment.